### PR TITLE
doc/md: create edges with mutation

### PR DESCRIPTION
**Pull request by giautm**
Signed-off-by: Giau. Tran Minh <hello@giautm.dev>

### CODE REVIEW

File: doc/md/tutorial-todo-gql-mutation-input.md [hunk 1]
```diff
@@ -177,4 +176,129 @@ mutation UpdateTodo {
     }
   }
 }
-```
\ No newline at end of file
+```
+
+## Create edges with mutations
+
+If you want to create the edges of a node in the same mutation, you can extend the GQL mutation input with the edge fields:
```
Comment (hilakashai) [inline doc/md/tutorial-todo-gql-mutation-input.md:183]:
```suggestion
To create the edges of a node in the same mutation, you can extend the GQL mutation input with the edge fields:
```

File: doc/md/tutorial-todo-gql-mutation-input.md [hunk 2]
```diff
@@ -177,4 +176,129 @@ mutation UpdateTodo {
...
+
+```graphql title="extended.graphql"
+extend input CreateTodoInput {
+  createChildren: [CreateTodoInput!]
+}
+```
+
+Then, you need to run code generation again:
```
Comment (hilakashai) [inline doc/md/tutorial-todo-gql-mutation-input.md:191]:
```suggestion
Next, run code generation again:
```

File: doc/md/tutorial-todo-gql-mutation-input.md [hunk 3]
```diff
@@ -177,4 +176,129 @@ mutation UpdateTodo {
...
+```go
+go generate .
+```
+
+GQLGen will generate the resolver for the `createChildren` field, and you can use it in your resolver:
```
Comment (hilakashai) [inline doc/md/tutorial-todo-gql-mutation-input.md:196]:
```suggestion
GQLGen will generate the resolver for the `createChildren` field, allowing you to use it in your resolver:
```

File: doc/md/tutorial-todo-gql-mutation-input.md [hunk 4]
```diff
@@ -177,4 +176,129 @@ mutation UpdateTodo {
...
+	ids := make([]int, len(todos))
+	for i := range todos {
+		ids[i] = todos[i].ID
+	}
+	obj.ChildIDs = append(obj.ChildIDs, ids...)
+	return nil
+}
+```
+
+Changes following lines to use the transactional client:
```
Comment (hilakashai) [inline doc/md/tutorial-todo-gql-mutation-input.md:233]:
```suggestion
Change the following lines to use the transactional client:
```

File: doc/md/tutorial-todo-gql-mutation-input.md [hunk 5]
```diff
@@ -177,4 +176,129 @@ mutation UpdateTodo {
...
+}
+
+// UpdateTodo is the resolver for the updateTodo field.
+func (r *mutationResolver) UpdateTodo(ctx context.Context, id int, input ent.UpdateTodoInput) (*ent.Todo, error) {
+	// highlight-next-line
+	return ent.FromContext(ctx).Todo.UpdateOneID(id).SetInput(input).Save(ctx)
+}
+```
+
+Testing our mutation with the children:
```
Comment (hilakashai) [inline doc/md/tutorial-todo-gql-mutation-input.md:249]:
```suggestion
Test the mutation with the children:
```

**Review by a8m — APPROVED** [review]
💯

**Pull request merged**